sheepdog/auth/__init__.py

```diff
@@ -113,9 +115,27 @@ def authorize_and_call(*args, **kwargs):
 def authorize(program, project, roles):
     resource = "/programs/{}/projects/{}".format(program, project)
     jwt = get_jwt_from_header()
-    authz = flask.current_app.auth.auth_request(
-        jwt=jwt, service="sheepdog", methods=roles, resources=[resource]
-    )
+    cache_key = str(hash((jwt, "sheepdog", tuple(roles), (resource))))
```

Comment:
i'm a bit unsure about using the hash function here, we shouldn't take any risks with authz checks and [from a quick google search there's a "high" risk of collision](https://stackoverflow.com/a/68048420). A long key that includes the whole JWT [worked fine for orthanc](https://github.com/uc-cdis/OrthancDocker/blob/f73b2ca27937def1f0102b996773f589fd4d80fc/orthanc-gen3/authz_filter.py#L41), could we do that here?

Reply:
Thanks for pointing this out! I didn’t realize hash was so prone to collisions. I’ll replace it with an underscore-separated string, similar to how it’s handled in Orthanc.